optaplanner-core/src/main/java/org/optaplanner/core/impl/score/buildin/bendable/BendableScoreInliner.java

```diff
@@ -17,30 +17,28 @@
 package org.optaplanner.core.impl.score.buildin.bendable;
 
 import java.util.Arrays;
-import java.util.function.Consumer;
+import java.util.List;
+import java.util.function.Supplier;
 
-import org.optaplanner.core.api.score.Score;
 import org.optaplanner.core.api.score.buildin.bendable.BendableScore;
 import org.optaplanner.core.impl.score.inliner.IntWeightedScoreImpacter;
 import org.optaplanner.core.impl.score.inliner.ScoreInliner;
 
 public class BendableScoreInliner extends ScoreInliner<BendableScore> {
 
-    private int[] hardScores;
-    private int[] softScores;
+    private final int[] hardScores;
+    private final int[] softScores;
 
     public BendableScoreInliner(boolean constraintMatchEnabled, int hardLevelsSize, int softLevelsSize) {
-        super(constraintMatchEnabled);
+        super(constraintMatchEnabled, BendableScore.zero(hardLevelsSize, softLevelsSize));
         hardScores = new int[hardLevelsSize];
         softScores = new int[softLevelsSize];
     }
 
     @Override
-    public IntWeightedScoreImpacter buildWeightedScoreImpacter(BendableScore constraintWeight) {
-        if (constraintWeight.equals(BendableScore.zero(hardScores.length, softScores.length))) {
-            throw new IllegalArgumentException("The constraintWeight (" + constraintWeight + ") cannot be zero,"
-                    + " this constraint should have been culled during node creation.");
-        }
+    public IntWeightedScoreImpacter buildWeightedScoreImpacter(String constraintPackage, String constraintName,
+            BendableScore constraintWeight) {
+        ensureNonZeroConstraintWeight(constraintWeight);
```

Comment:
Naming consistency: if that ensureNon... method throws an exception if it's not the case, it should be called assertNon... instead.